manager/orchestrator/update/updater.go

```diff
@@ -423,6 +428,14 @@ func (u *Updater) updateTask(ctx context.Context, slot orchestrator.Slot, update
 				u.updatedTasksMu.Lock()
 				u.updatedTasks[updated.ID] = time.Now()
 				u.updatedTasksMu.Unlock()
+
+				if startThenStop {
+					_, err := u.store.Batch(func(batch *store.Batch) error {
+						u.removeOldTasks(ctx, batch, slot)
```

Comment:
I don't think it's necessary to return an error because the update has already happened. If the old task couldn't be found or was already shut down, this is a non-fatal condition. I'll add a warning log, though.